fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -457,8 +457,14 @@ private long getTimestamp(final String property, final long created) throws Repo
         LOGGER.trace("Using {} date", property);
         final long timestamp = getProperty(property).getDate().getTimeInMillis();
         if (timestamp < created && created > NO_TIME && !isFrozenResource()) {
-            LOGGER.trace("Updating {} with later created date", property);
-            getNode().setProperty(property, created);
+            try {
+                LOGGER.trace("Updating {} with later created date", property);
+                getNode().setProperty(property, created);
+            } catch (final javax.jcr.AccessDeniedException e) {
+                LOGGER.trace("Unable to update {} with the later created date", property);
+            } catch (final javax.jcr.RepositoryException e) {
+                LOGGER.warn("Unable to update " + property + " with the later created date", e);
```

Comment:
It seems like the two primary options are:
1. Ensuring the dates are correct at the point of resource creation, which will require an extra `save` on create.
2. Do not update the property on subsequent `getTimestamp()` requests, but instead return the potentially adjusted value.

From a performance perspective, (2) is preferred. From a data integrity perspective, (1) is preferred. How do you all feel about going with option (2) with a `debug` statement that clearly indicates the two dates and that we are returning one that is adjusted?

Like @mikedurbin, I am not a fan of updating resources on HTTP GET requests.